src/test/java/io/camunda/testing/assertions/JobAssertTest.java

```diff
@@ -0,0 +1,285 @@
+
+      // then
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      assertThat(actual).hasBpmnProcessId(PROCESS_INSTANCE_ID);
+    }
+
+    @Test
+    void testHasRetries() throws InterruptedException {
+      // given
+      deployProcess(PROCESS_INSTANCE_BPMN);
+      final Map<String, Object> variables = Collections.singletonMap("totalLoops", 1);
+      startProcessInstance(PROCESS_INSTANCE_ID, variables);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse =
+          client.newActivateJobsCommand().jobType("test").maxJobsToActivate(1).send().join();
+
+      // then
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      assertThat(actual).hasRetries(1);
+    }
+
+    @Test
+    void testExtractingVariables() throws InterruptedException {
+      // given
+      deployProcess(PROCESS_INSTANCE_BPMN);
+      final Map<String, Object> variables = Collections.singletonMap("totalLoops", 1);
+      startProcessInstance(PROCESS_INSTANCE_ID, variables);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse =
+          client.newActivateJobsCommand().jobType("test").maxJobsToActivate(1).send().join();
+
+      // then
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      assertThat(actual)
+          .extractingVariables()
+          .containsOnly(entry("totalLoops", 1), entry("loopAmount", 0));
+    }
+
+    @Test
+    void testExtractingHeaders() throws InterruptedException {
+      // given
+      deployProcess(PROCESS_INSTANCE_BPMN);
+      final Map<String, Object> variables = Collections.singletonMap("totalLoops", 1);
+      startProcessInstance(PROCESS_INSTANCE_ID, variables);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse =
+          client.newActivateJobsCommand().jobType("test").maxJobsToActivate(1).send().join();
+
+      // then
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      assertThat(actual).extractingHeaders().isEmpty();
+    }
+  }
+
+  // These tests are just for assertion testing purposes. These should not be used as examples.
+  @Nested
+  class UnhappyPathTests {
+    private RecordStreamSource recordStreamSource;
+
+    @Test
+    void testHasElementIdFailure() throws InterruptedException {
+      // given
+      deployProcess(PROCESS_INSTANCE_BPMN);
+      final Map<String, Object> variables = Collections.singletonMap("totalLoops", 1);
+      startProcessInstance(PROCESS_INSTANCE_ID, variables);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse =
+          client.newActivateJobsCommand().jobType("test").maxJobsToActivate(1).send().join();
+
+      // then
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+
+      assertThatThrownBy(() -> assertThat(actual).hasElementId(WRONG_VALUE))
+          .isInstanceOf(AssertionError.class)
+          .hasMessageContainingAll(ELEMENT_ID, WRONG_VALUE);
```

Comment:
🙃 This doesn't properly test the assertion message. The expected and actual could be flipped in the message and this test will still succeed. This is the same in the other Failure assertions.